tests/integration/steps/validation_requests.go

```diff
@@ -70,6 +71,25 @@ func ValidateResponseStatusCode(ctx context.Context, path string, expectedCode i
 	return ctx, c.Get(url, asserter)
 }
 
+func ValidateHeaderStatusCode(ctx context.Context, givenHeaderName, givenHeaderValue, path string, expectedCode int) (context.Context, error) {
```

Comment:
I find the name a little bit confusing. As far as I understand we only want to validate the response code, but we also send headers with the request. The function names suggests that we are also validating headers.